test/test_path_sanitization.rb

```diff
@@ -28,4 +28,35 @@ class TestPathSanitization < JekyllUnitTest
     assert_equal source_dir("files", "hi.txt"),
                  Jekyll.sanitized_path(source_dir, "f./../../../../../../files/hi.txt")
   end
+
+  context "on Windows with absolute path" do
+    setup do
+      @base_path = "D:/demo"
+      @file_path = "D:/demo/_site"
+      allow(Dir).to receive(:pwd).and_return("D:/")
+    end
+
+    should "strip just the clean path drive name" do
+      assert_equal "D:/demo/_site",
+                   Jekyll.sanitized_path(@base_path, @file_path)
+    end
+  end
+
+  context "on Windows with file path has matching prefix" do
+    setup do
+      @base_path = "D:/site"
+      @file_path = "D:/sitemap.xml"
+      allow(Dir).to receive(:pwd).and_return("D:/")
+    end
+
+    should "not strip base path" do
+      assert_equal "D:/site/sitemap.xml",
```

Comment:
I'm still unclear on when this test's behavior is desirable. It seems like the path to `sitemap.xml` is absolute, so under what condition does it make sense to transform it into a relative path that is appended on to the base path? I understand that this is largely just testing existing behavior, but I'm wondering if that behavior is needed.